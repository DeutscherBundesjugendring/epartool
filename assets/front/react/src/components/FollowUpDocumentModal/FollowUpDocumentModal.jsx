import React from 'react';
import moment from 'moment';
import EmbeddedVideo from '../EmbeddedVideo/EmbeddedVideo';
import Reputation from '../Reputation/Reputation';
import RaisedButton from '../RaisedButton/RaisedButton';


const FollowDocumentModal = (props) => {
  let glypClasses = 'followup-type-icon glyphicon';
  let isType = false;

  if (props.type === 'supporting') {
    glypClasses += ' glyphicon-heart';
    isType = true;
  } else if (props.type === 'action') {
    glypClasses += ' glyphicon-play';
    isType = true;
  } else if (props.type === 'rejected') {
    glypClasses += ' glyphicon-minus-sign';
    isType = true;
  } else if (props.type === 'end') {
    glypClasses += ' glyphicon-lock';
    isType = true;
  }

  return (<div>
    <div className="modal-backdrop fade in" />
    <div
      className="modal fade in"
      aria-hidden="true"
      aria-labelledby="modalFollowupLabel"
      style={{ display: 'block' }}
    >
      <div className="modal-dialog">
        <div className="modal-content">

          <div className="modal-header">
            <button
              type="button"
              className="close"
              aria-label="Close"
              onTouchTap={props.closeAction}
            >
              <span aria-hidden="true">Ã—</span>
            </button>
            <h4 className="modal-title" id="modalFollowupLabel">
              {props.title}
            </h4>
          </div>
          <div className="modal-body">

            <div className="well well-accent followup-type-wrap">
              {isType && (
                <div className="followup-type followup-type-right">
                  <span className={glypClasses} aria-hidden="true" />
                </div>
              )}
              <div className="row">
                <div className="col-sm-3">
                  <img
                    className="img-responsive pull-left"
                    src={props.previewImageLink}
                    alt={props.title}
                  />
                </div>
                <div className="col-sm-8">
                  <p>
                    {props.author}
                  </p>
                  <p className="small">
                    {props.dateMonthYearOnly
                      ? moment(props.date).format('MMMM YYYY')
                      : moment(props.date).format('D MMMM YYYY')
                    }
                  </p>
                  <button
                    className="btn btn-default btn-default-alt"
                    onTouchTap={props.downloadAction}
                  >
                    <span className="glyphicon glyphicon-file icon-offset" />
                    {props.downloadLabel}
                  </button>
                </div>
              </div>
            </div>

            <hr />

            {!!props.snippets && props.snippets.map((snippet, index) => (
              <div
                className="well well-simple well-simple-light text-left well-heading"
                key={index}
              >
                {!!snippet.videoService && !!snippet.videoId &&
                  <EmbeddedVideo
                    videoService={snippet.videoService}
                    videoId={snippet.videoId}
                  />
                }

                <div dangerouslySetInnerHTML={{ __html: snippet.snippetExplanation }} />

                <div className="text-right offset-top-small offset-bottom-small">
                  <span className="offset-right">
                    <Reputation
                      likeCount={snippet.likeCount}
                      dislikeCount={snippet.dislikeCount}
                      likeAction={snippet.likeAction}
                      dislikeAction={snippet.dislikeAction}
                    />
                  </span>
                  {snippet.showFollowPathButton &&
                    <RaisedButton
                      label={snippet.followPathLabel}
                      onTouchTap={snippet.followPathAction}
                    />
                  }
                </div>
              </div>
            ))}
          </div>
        </div>
      </div>
    </div>
  </div>
  );
};

FollowDocumentModal.propTypes = {
  type: React.PropTypes.string.isRequired,
  title: React.PropTypes.string.isRequired,
  author: React.PropTypes.string.isRequired,
  date: React.PropTypes.objectOf(Date).isRequired,
  dateMonthYearOnly: React.PropTypes.bool.isRequired,
  previewImageLink: React.PropTypes.string.isRequired,
  downloadAction: React.PropTypes.func.isRequired,
  downloadLabel: React.PropTypes.string.isRequired,
  closeAction: React.PropTypes.func.isRequired,
  snippets: React.PropTypes.arrayOf(React.PropTypes.shape({
    snippetExplanation: React.PropTypes.string.isRequired,
    likeAction: React.PropTypes.func.isRequired,
    likeCount: React.PropTypes.number.isRequired,
    dislikeAction: React.PropTypes.func.isRequired,
    dislikeCount: React.PropTypes.number.isRequired,
    followPathAction: React.PropTypes.func.isRequired,
    followPathLabel: React.PropTypes.string.isRequired,
    showFollowPathButton: React.PropTypes.bool.isRequired,
  })),
};

export default FollowDocumentModal;
