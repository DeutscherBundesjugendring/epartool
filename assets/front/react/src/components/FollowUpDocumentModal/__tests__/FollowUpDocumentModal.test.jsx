import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import sinon from 'sinon';
import React from 'react';
import injectTapEventPlugin from 'react-tap-event-plugin';
import FollowUpDocumentModal from '../FollowUpDocumentModal';


injectTapEventPlugin();

const element = (
  <FollowUpDocumentModal
    title="Document title"
    author="Author of document"
    date={new Date()}
    dateMonthYearOnly={false}
    previewImageLink="http://www.example.com/image.jpg"
    downloadAction={() => {}}
    downloadLabel="Download document"
    closeAction={() => {}}
  />
);

const snippetElement = {
  snippetExplanation: 'Snippet',
  likeAction: () => {},
  likeCount: 0,
  dislikeAction: () => {},
  dislikeCount: 0,
  followPathAction: () => {},
  followPathLabel: 'Follow path',
};


describe('rendering', () => {
  it('renders correctly with long date without snippets', () => {
    const tree = shallow(
      React.cloneElement(element)
    );

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly with short date without snippets', () => {
    const tree = shallow(
      React.cloneElement(element, { dateMonthYearOnly: true })
    );

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly with one snippet', () => {
    const tree = shallow(
      React.cloneElement(element, { snippets: [
        Object.assign({}, snippetElement),
      ] })
    );

    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});

describe('functionality', () => {
  it('calls close action', () => {
    const spy = sinon.spy();
    const component = shallow(
      React.cloneElement(element, { closeAction: spy })
    );

    component.find('button').first().simulate('touchTap', { stopPropagation: () => {} });
    expect(spy.calledOnce).toEqual(true);
  });

  it('calls download action', () => {
    const spy = sinon.spy();
    const component = shallow(
      React.cloneElement(element, { downloadAction: spy })
    );

    component.find('button').last().simulate('touchTap', { stopPropagation: () => {} });
    expect(spy.calledOnce).toEqual(true);
  });

  it('calls snippet like action', () => {
    const spy = sinon.spy();
    const snippetElementCopy = Object.assign({}, snippetElement);
    snippetElementCopy.likeAction = spy;

    const component = shallow(
      React.cloneElement(element, { snippets: [snippetElementCopy] })
    );

    component.find('ThumbButton').first().simulate('touchTap');
    expect(spy.calledOnce).toEqual(true);
  });

  it('calls snippet dislike action', () => {
    const spy = sinon.spy();
    const snippetElementCopy = Object.assign({}, snippetElement);
    snippetElementCopy.dislikeAction = spy;

    const component = shallow(
      React.cloneElement(element, { snippets: [snippetElementCopy] })
    );

    component.find('ThumbButton').last().simulate('touchTap');
    expect(spy.calledOnce).toEqual(true);
  });

  it('calls snippet follow path action', () => {
    const spy = sinon.spy();
    const snippetElementCopy = Object.assign({}, snippetElement);
    snippetElementCopy.followPathAction = spy;

    const component = shallow(
      React.cloneElement(element, { snippets: [snippetElementCopy] })
    );

    component.find('RaisedButton').simulate('touchTap');
    expect(spy.calledOnce).toEqual(true);
  });
});
