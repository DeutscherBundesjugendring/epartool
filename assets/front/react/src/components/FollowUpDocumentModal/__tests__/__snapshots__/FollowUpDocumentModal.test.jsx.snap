// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly with long date without snippets 1`] = `
<div>
  <div
    className="modal-backdrop fade in"
  />
  <div
    aria-hidden="true"
    aria-labelledby="modalFollowupLabel"
    className="modal fade in"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div
      className="modal-dialog"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <button
            aria-label="Close"
            className="close"
            onTouchTap={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
          <h4
            className="modal-title"
            id="modalFollowupLabel"
          >
            Document
          </h4>
        </div>
        <div
          className="modal-body"
        >
          <div
            className="well well-accent followup-type-wrap"
          >
            <div
              className="row"
            >
              <div
                className="col-sm-3 text-center"
              >
                <img
                  alt="Document title"
                  className="img-responsive center-block offset-bottom"
                  src="http://www.example.com/image.jpg"
                />
                <button
                  className="btn btn-default btn-default-alt"
                  onTouchTap={[Function]}
                >
                  <span
                    className="glyphicon glyphicon-file icon-offset"
                  />
                  Download document
                </button>
              </div>
              <div
                className="col-sm-8"
              >
                <h4
                  className="well-title"
                >
                  Document title
                </h4>
                <p>
                  Author of document
                </p>
                <p
                  className="small"
                >
                  1 January 2017
                </p>
              </div>
            </div>
          </div>
          <hr />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`rendering renders correctly with one snippet 1`] = `
<div>
  <div
    className="modal-backdrop fade in"
  />
  <div
    aria-hidden="true"
    aria-labelledby="modalFollowupLabel"
    className="modal fade in"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div
      className="modal-dialog"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <button
            aria-label="Close"
            className="close"
            onTouchTap={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
          <h4
            className="modal-title"
            id="modalFollowupLabel"
          >
            Document
          </h4>
        </div>
        <div
          className="modal-body"
        >
          <div
            className="well well-accent followup-type-wrap"
          >
            <div
              className="row"
            >
              <div
                className="col-sm-3 text-center"
              >
                <img
                  alt="Document title"
                  className="img-responsive center-block offset-bottom"
                  src="http://www.example.com/image.jpg"
                />
                <button
                  className="btn btn-default btn-default-alt"
                  onTouchTap={[Function]}
                >
                  <span
                    className="glyphicon glyphicon-file icon-offset"
                  />
                  Download document
                </button>
              </div>
              <div
                className="col-sm-8"
              >
                <h4
                  className="well-title"
                >
                  Document title
                </h4>
                <p>
                  Author of document
                </p>
                <p
                  className="small"
                >
                  1 January 2017
                </p>
              </div>
            </div>
          </div>
          <hr />
          <div
            className="well well-simple well-simple-light text-left well-heading"
          >
            <div
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Snippet",
                }
              }
            />
            <div
              className="offset-top-small offset-bottom-small"
            >
              <div
                className="row"
              >
                <div
                  className="col-sm-6 text-left"
                >
                  <div
                    className="offset-top-small"
                  >
                    <Reputation
                      dislikeAction={[Function]}
                      dislikeCount={0}
                      likeAction={[Function]}
                      likeCount={0}
                      voted={false}
                    />
                  </div>
                </div>
                <div
                  className="col-sm-6 text-right"
                >
                  <div
                    className="offset-top-small"
                  >
                    <RaisedButton
                      disabled={false}
                      label="Follow path"
                      onTouchTap={[Function]}
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`rendering renders correctly with short date without snippets 1`] = `
<div>
  <div
    className="modal-backdrop fade in"
  />
  <div
    aria-hidden="true"
    aria-labelledby="modalFollowupLabel"
    className="modal fade in"
    style={
      Object {
        "display": "block",
      }
    }
  >
    <div
      className="modal-dialog"
    >
      <div
        className="modal-content"
      >
        <div
          className="modal-header"
        >
          <button
            aria-label="Close"
            className="close"
            onTouchTap={[Function]}
            type="button"
          >
            <span
              aria-hidden="true"
            >
              ×
            </span>
          </button>
          <h4
            className="modal-title"
            id="modalFollowupLabel"
          >
            Document
          </h4>
        </div>
        <div
          className="modal-body"
        >
          <div
            className="well well-accent followup-type-wrap"
          >
            <div
              className="row"
            >
              <div
                className="col-sm-3 text-center"
              >
                <img
                  alt="Document title"
                  className="img-responsive center-block offset-bottom"
                  src="http://www.example.com/image.jpg"
                />
                <button
                  className="btn btn-default btn-default-alt"
                  onTouchTap={[Function]}
                >
                  <span
                    className="glyphicon glyphicon-file icon-offset"
                  />
                  Download document
                </button>
              </div>
              <div
                className="col-sm-8"
              >
                <h4
                  className="well-title"
                >
                  Document title
                </h4>
                <p>
                  Author of document
                </p>
                <p
                  className="small"
                >
                  January 2017
                </p>
              </div>
            </div>
          </div>
          <hr />
        </div>
      </div>
    </div>
  </div>
</div>
`;
